### ISSUE TITLE
Cutscene ending

### ISSUE THREAD

**Issue by Xcelled** [aura-project/aura#72]
If the party leader lets the cutscene run to completion normally, other party members still get the "Your party leader has skipped this clip..." message, and are presumably cut off even if they're not done with the cutscene.

**Comment by Xcelled** [comment]
Funnily enough, pressing escape doesn't let you out of the cutscene prematurely either. I just noticed this as a non-leader.

**Comment by exectails** [comment]
>Funnily enough, pressing escape doesn't let you out of the cutscene prematurely either. I just noticed this as a non-leader.

That's the way it's supposed to be, isn't it? I'm pretty sure you weren't able to cancel cutscenes if the party leader was still watching. Or... was that only for "important" cutscenes? Like in mainstream dungeons?

**Comment by QewQew** [comment]
Correct, many cutscenes will allow you to skip the cutscene without waiting for the leader.
It's generally those that are important that cannot be skipped without the leader.

**Comment by exectails** [comment]
Hm, some cutscenes have an attribute named `terminated_by_key`, which the cutscenes in the G1 file don't have. I guess that might be the deciding factor here. So anybody can cancel any cutscene for himself, as long as that option isn't set.

The other problem is that you usually get the "skipped" message. The reason for this is that there are no separate cancel and finish packets, in both cases you get `FinishedCutscene`, which in turn has to cancel the cutscene for everybody, which results in the skipped message, because it's canceled for the others.

In light of that new option mentioned above, I assume that this is the case on officials as well, but you generally don't notice it. If the option is set, you're the only one with control over your cutscene. If it's not set, the party leader is in control. In these cutscenes you usually have dialogues, and won't finish at the same time as the leader. You'll either be faster, and get the "wait" msg, or you're slower, and the cutscene *will* be skipped for you.

**Comment by Xcelled** [comment]
I'm fairly certain you don't get the message, but I can run a quick Alby later to test.

**Comment by exectails** [comment]
>Isn't there some packet sent when you hit esc?

Yea, but that packet has never been consistent I think, during certain times we got it every time escape was pressed, during others we didn't get it at all. I wouldn't want to couple cutscenes to that packet without knowing its purpose and whether we can trust it. Not to mention, getting the escape packet wouldn't guarantee that someone just *canceled* the cutscene.

But thinking about it again, *can* you even cancel the "important" cutscenes? Like at all? I don't remember ever canceling the story cutscenes, but I did "skip" some, rapidly clicking on the dialog boxes. This would make the `terminated_by_key` attribute seem very logical. It's not about important or not, leader or member, it's about canceling all together.

**Comment by Xcelled** [comment]
I know in a normal dungeon, you can cancel boss cutscenes early as any member. If you're the leader, cancelling your cutscene will also cancel everyone else's with the message "Your party leader has skipped the clip". I remember specifically in the G1 Final boss cutscene, only the party leader could cancel the cutscene. Everyone else had to wait. In Iria, Saga, no cutscenes may be skipped *at all* the first time through.

**Comment by QewQew** [comment]
>In Iria, Saga, no cutscenes may be skipped at all the first time through.

The client handles that part, however; through mods, you could achieve skipping.

**Comment by exectails** [comment]
> If you're the leader, cancelling your cutscene will also cancel everyone else's

Are you really sure about that...? We'll need research =/ Preferably logs of two people running a dungeon, with both logging. One log with the leader canceling, one with the player canceling, and one without cancellation.

**Comment by DeleteIT** [comment]
I'm pretty certain that only the dungeons like Math Adv had that leader skip, and it was a case of being *unable* to skip the scene as a regular member.

In dungeons like Ciar Normal/basic, members were left to finish cut scenes as they pleased, regardless of leader skipping.

**Comment by exectails** [comment]
Note to whoever will work on this (aka me): The PartyLeader reference in Dungeon needs to be updated ones the cut scenes have been updated, as all dungeon cut scenes currently depend on a valid party leader creature. That won't break anything, but it needs fixing, otherwise the cut scene won't play properly if they're missing.
